Source/WTF/wtf/SuspendableWorkQueue.cpp

```diff
@@ -26,33 +26,53 @@
 #include "config.h"
 #include <wtf/SuspendableWorkQueue.h>
 
+#include <wtf/Logging.h>
+
 namespace WTF {
 
-Ref<SuspendableWorkQueue> SuspendableWorkQueue::create(const char* name, WorkQueue::QOS qos)
+Ref<SuspendableWorkQueue> SuspendableWorkQueue::create(const char* name, WorkQueue::QOS qos, ShouldLog shouldLog)
 {
-    return adoptRef(*new SuspendableWorkQueue(name, qos));
+    return adoptRef(*new SuspendableWorkQueue(name, qos, shouldLog));
 }
 
-SuspendableWorkQueue::SuspendableWorkQueue(const char* name, QOS qos)
+SuspendableWorkQueue::SuspendableWorkQueue(const char* name, QOS qos, ShouldLog shouldLog)
     : WorkQueue(name, qos)
+    , m_shouldLog(shouldLog)
 {
     ASSERT(isMainThread());
 }
 
+void SuspendableWorkQueue::log(ASCIILiteral message)
```

Comment:
Please use `RELEASE_LOG_IF(m_shouldLog)` instead of this whole function.
You can also create your own new RELEASE_LOG_SWQ() macro in this file which wraps the  `RELEASE_LOG_IF(m_shouldLog)` call to simplify call sites.

Release logging should be cheap and thus should avoid function calls.